exec.go

```diff
@@ -455,10 +456,23 @@ func (db *DB) Save(table string, data interface{}) error {
 		return db.saveRow(table, data)
 	} else {
 		for i := 0; i < rv.Len(); i++ {
-			err = db.saveRow(table, rv.Index(i).Interface())
-			if err != nil {
-				return err
+
+			if rv.Index(i).Elem().Type().Kind() == reflect.Struct && rv.Index(i).Type().Kind() == reflect.Interface {
```

Comment:
added a comment, when structs are passed inside slice, `Index(i)` returns struct packed into interface that needs to be unpacked

Reply:
What test are you fixing with this? Removing this whole block still passes **TestInterfaceSliceSaveStructs** (with your data copy patch). But the same effect yields if you simple skip _canAddr_ in _reflect.struct_ case in "checkData". May be you can provide me with an idea why we need this block?

Reply:
with my new update of not running `setPrimaryKey` if the field not addressable, it will still work without this block, this block just ensures to pass the addressable reference to `db.saveRow` so that the primary key could be set whenever needed